develop/addons/schema-driven-forms/customising-form-behaviour/validation.rst

```diff
@@ -129,7 +129,105 @@ For example:
     ...
 
 
-    @form.validator.validator(field=IPizzaOrder['phone_number'])
+    class IPizzaOrder(model.Schema):
+
+        phone_number = schema.TextLine(
+            title=_(u'Phone number'),
+            description=_(
+                u'Your phone number in international format. '
+                u'E.g. +44 12 123 1234'
+            ),
+            required=False,
+            default=u'')
+
+
+    @form.validator(field=IPizzaOrder['phone_number'])
+    def validate_phone_number(value):
+        """Validate international phone number on input
+        """
+
+        allowed_characters = '+- () / 0123456789'
+
+        if value != None:
+            value = value.strip()
+
+            if value == '':
+                # Assume empty string = no input
+                return
+
+            # The value is not required
+            for c in value:
+                if c not in allowed_characters:
+                    raise zope.interface.Invalid(
+                        _(u'Phone number contains bad characters')
+                    )
+
+            if len(value) < 7:
+                raise zope.interface.Invalid(_(u'Phone number is too short'))
+
+
+The *@form.validator.validator()* decorator registers a validator adapter. When
+the validation is invoked, the decorated function will be called with
+the field’s value as an argument and given an opportunity to raise a
+validation error, much like the constraint above. Again like the
+constraint, the default validator is called first, so things like the
+required flag and indeed any custom constraint are processed first.
+
+The *@form.validator.validator()* decorator can take keyword arguments to make the
+validator more specific or more generic. The valid values are:
+
+context
```

Comment:
Please insert a blank line before each entry in a definition list for readability. Also each definition definition should be one sentence per line, maintaining indentation.